### PR TITLE
Enable saving as zip

### PR DESCRIPTION
**Pull request by wuhu**
#142

### CODE REVIEW

File: padl/transforms.py [hunk 1]
```diff
@@ -165,13 +168,41 @@ def pd_post_load(self, path: Path, i: int):
         :param i: Unique transform index, can be used to construct filenames.
         """
 
+    def pd_zip_save(self, path: Union[Path, str], force_overwrite: bool = False):
+        """Save the transform to a zip-file at *path*.
+
+        The file's name should end with '.padl'. If no extension is given, it will be added
```
Comment (sjrl) [inline padl/transforms.py:174]: Should we enforce that the suffix is `.padl`?

File: padl/transforms.py [hunk 2]
```diff
@@ -1723,6 +1761,15 @@ def load(path):
     return transform
 
 
+def _zip_load(path: Union[Path, str]):
+    """Load a transform from a compressed '.padl' file. """
+    # we can't use a context because the files need to exist when using / saving again
```
Comment (sjrl) [inline padl/transforms.py:1766]: What do you mean by this comment exactly? It looks like you are using a `with` context below?

File: padl/transforms.py [hunk 3]
```diff
@@ -165,13 +168,41 @@ def pd_post_load(self, path: Path, i: int):
...
+
+        if path.exists():
+            if not force_overwrite:
+                raise FileExistsError(f'{path} exists, call with *force_overwrite* to overwrite.')
+            try:
+                rmtree(path)
+            except NotADirectoryError:
+                remove(path)
+
+        with TemporaryDirectory('.padl') as dirname:
```
Comment (sjrl) [inline padl/transforms.py:192]: We should consider making the `TemporaryDirectory` inside of `path`. People may not have write permissions in other parts of the machine.

File: padl/transforms.py (hunk 2 above)
Comment (wuhu) [reply padl/transforms.py:1766]: When using the context, the temporary directory is deleted when leaving. This doesn't work unfortunately as the files are needed for code inspection / saving. Thus we can't use the context.

Comment (sjrl) [reply padl/transforms.py:1766]: Could you add a note, specifying that the context we can't use refers to the `TemporaryDirectory`?

File: padl/transforms.py [hunk 4]
```diff
@@ -1692,20 +1723,27 @@ def __call__(self, args):
...
+    if compress:
+        transform.pd_zip_save(path, force_overwrite)
+    else:
+        transform.pd_save(path, force_overwrite)
 
 
 def load(path):
     """Load a transform (as saved with padl.save) from *path*. """
+    if Path(path).is_file():
+        return zip_load(path)
```
Comment (sjrl) [inline padl/transforms.py:1746]: Should this be calling `_zip_load`?

**Comment by sjrl** [comment]
Could you add a unittest?

**Pull request merged**